### PR TITLE
Remove some unnecessary defaults and make a couple of tidy up after the SASL work.

### PR DESCRIPTION
**Pull request by k-wall**
### Type of change

_Select the type of your PR_

- Refactoring

### Description

Remove some unnecessary defaults and a couple of tidy up after the SASL work.

_I was expecting to require a few more changes to enable SASL OAuth through the extensions.  It turned out not to be
case.  I made a couple of tidy ups on the way to discovering that, resulting in this PR._

### Additional Context

_Why are you making this pull request?_

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

**Comment by k-wall** [comment]
I'm drafting out an intergration test in Kroxylicious to verify the use-case.

File: impl/src/main/java/io/kroxylicious/testing/kafka/common/ConstraintUtils.java [hunk 1]
```diff
@@ -177,12 +178,18 @@ private static <A extends Annotation> A mkAnnotation(Class<A> annoType, Map<Stri
                 annoType.getDeclaredMethod(member);
             }
             catch (NoSuchMethodException e) {
-                throw new RuntimeException(e);
+                throw new IllegalArgumentException("Annotation %s does not have an element %s".formatted(annoType.getName(), member));
             }
         }
-        Objects.requireNonNull(members);
+        var defaults = Arrays.stream(annoType.getDeclaredMethods())
```
Comment (k-wall) [inline impl/src/main/java/io/kroxylicious/testing/kafka/common/ConstraintUtils.java:184]: Previous the proxy wasn't respecting the annotation defaults.  I had a workaround in the production code for the test shortcoming.  This change addresses the test shortcoming.

File: impl/src/main/java/io/kroxylicious/testing/kafka/common/KafkaClusterConfig.java [hunk 2]
```diff
@@ -205,7 +205,7 @@ else if (annotation instanceof Version version) {
             }
             else if (annotation instanceof SaslMechanism mechanism) {
                 useSasl = true;
-                builder.saslMechanism(Optional.ofNullable(mechanism.value()).orElse(PLAIN_SASL_MECHANISM_NAME));
```
Comment (k-wall) [inline impl/src/main/java/io/kroxylicious/testing/kafka/common/KafkaClusterConfig.java:208]: that's the unnecessary defaulting - it was only ever null when coming from the test proxy.

File: impl/src/main/java/io/kroxylicious/testing/kafka/common/KafkaClusterConfig.java [hunk 3]
```diff
@@ -66,13 +66,13 @@ public class KafkaClusterConfig {
...
-    public static final String BROKER_ROLE = "broker";
-    public static final String CONTROLLER_ROLE = "controller";
+    private static final String BROKER_ROLE = "broker";
+    private static final String CONTROLLER_ROLE = "controller";
 
-    public static final String CONTROLLER_LISTENER_NAME = "CONTROLLER";
-    public static final String EXTERNAL_LISTENER_NAME = "EXTERNAL";
-    public static final String INTERNAL_LISTENER_NAME = "INTERNAL";
-    public static final String ANON_LISTENER_NAME = "ANON";
+    private static final String CONTROLLER_LISTENER_NAME = "CONTROLLER";
```
Comment (SamBarker) [inline impl/src/main/java/io/kroxylicious/testing/kafka/common/KafkaClusterConfig.java:72]: 🎉

**Review by SamBarker — APPROVED** [review]
LGTM thanks for cleaning it up @k-wall

**Pull request merged**